pkg/gardenlet/controller/shoot/seed_registration_control.go

```diff
@@ -172,675 +130,234 @@ func (c *defaultSeedRegistrationControl) Reconcile(shootObj *gardencorev1beta1.S
-				clusterRoleBinding.RoleRef = rbacv1.RoleRef{
-					APIGroup: "rbac.authorization.k8s.io",
-					Kind:     "ClusterRole",
-					Name:     bootstraputil.GardenerSeedBootstrapper,
-				}
-				clusterRoleBinding.Subjects = []rbacv1.Subject{
-					{
-						Kind:      "ServiceAccount",
-						Name:      saName,
-						Namespace: v1beta1constants.GardenNamespace,
+		// Initialize gardenlet configuraton and parameters
+		gardenlet = &seedmanagementv1alpha1.Gardenlet{
+			Config: &configv1alpha1.GardenletConfiguration{
+				Resources:    resources,
+				FeatureGates: shootedSeed.FeatureGates,
+				SeedConfig: &configv1alpha1.SeedConfig{
+					Seed: gardencorev1beta1.Seed{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: shoot.Labels,
+						},
+						Spec: *seedSpec,
 					},
-				}
-				return nil
-			}); err != nil {
-				return err
-			}
-
-			bootstrapKubeconfig, err = bootstraputil.MarshalKubeconfigWithToken(&restConfig, string(saSecret.Data[corev1.ServiceAccountTokenKey]))
-			if err != nil {
-				return err
-			}
-		} else {
-			// create bootstrap token with bootstrap kubeconfig in order to create CSR
-			var (
-				tokenID               = utils.ComputeSHA256Hex([]byte(shoot.Name))[:6]
-				validity              = 24 * time.Hour
-				refreshBootstrapToken = true
-				bootstrapTokenSecret  *corev1.Secret
-			)
-
-			secret := &corev1.Secret{}
-			if err := gardenClient.Client().Get(ctx, kutil.Key(metav1.NamespaceSystem, bootstraptokenutil.BootstrapTokenSecretName(tokenID)), secret); client.IgnoreNotFound(err) != nil {
-				return err
-			}
-
-			if expirationTime, ok := secret.Data[bootstraptokenapi.BootstrapTokenExpirationKey]; ok {
-				t, err := time.Parse(time.RFC3339, string(expirationTime))
-				if err != nil {
-					return err
-				}
-
-				if !t.Before(metav1.Now().UTC()) {
-					bootstrapTokenSecret = secret
-					refreshBootstrapToken = false
-				}
-			}
-
-			if refreshBootstrapToken {
-				bootstrapTokenSecret, err = kutil.ComputeBootstrapToken(ctx, gardenClient.Client(), tokenID, fmt.Sprintf("A bootstrap token for the Gardenlet for shooted seed %q.", shoot.Name), validity)
-				if err != nil {
-					return err
-				}
-			}
-
-			bootstrapKubeconfig, err = bootstraputil.MarshalKubeconfigWithToken(&restConfig, kutil.BootstrapTokenFrom(bootstrapTokenSecret.Data))
-			if err != nil {
-				return err
+				},
+			},
+			GardenConnectionBootstrap: &gardenConnectionBootstrap,
+			// For backward compatibility with the use-as-seed annotation this should rather be nil,
+			// since the seed registration controller implementation always inherits the seed connection settings
+			// from the parent gardenlet. However, I believe this really only works with "service account" anyway,
+			// and always fails with local dev setups since there is a wrong kubeconfig set in the parent.
+			SeedConnection:    &seedConnection,
+			MergeParentConfig: true,
```

Comment:
Seeing this I would opt for breaking compatibility with the Shooted seed annotation. We could get rid of both GardenConnectionBootstrap (always use bootstrap secret)  and SeedConnection (always use service account).
However, I do not know the original reason why there was ever the possibility to define multiple options for GardenConnectionBootstrap.

Reply:
I tend to disagree, see my comments above. 
* I would agree for removing `SeedConnection` if we hardcode the mechanism to "service account" (the only one that works with a local development setup). 
* I would agree for removing `GardenConnectionBootstrap`, but then we should always use not just a bootstrap secret, but a concrete kind of bootstrap secret, either "service account" or "bootstrap token". As I am also a bit unsure why both these 2 options exist, I guess this should be clarified first. @timebertt @rfranzke, do you know the reason?

Reply:
As mentioned above, we decide to remove `SeedConnection` but not `GardenConnectionBootstrap`.